### PR TITLE
Added support for Uncompressed Key

### PR DESCRIPTION
**Pull request by zktony**
## Describe your changes
Added support for Uncompressed Key.

### CODE REVIEW

File: pallets/thea/src/lib.rs [hunk 1]
```diff
@@ -746,19 +748,47 @@ impl<T: Config> Pallet<T> {
...
-					// Update nonce
-					<OutgoingNonce<T>>::insert(message.network, message.nonce);
-					<OutgoingMessages<T>>::insert(message.network, message.nonce, message);
+			let mut uncompressed_keys = vec![];
+			for public_key in queued.clone().into_iter() {
+				let public_key: sp_core::ecdsa::Public = public_key.into();
+				if let Ok(compressed_key) = libsecp256k1::PublicKey::parse_compressed(&public_key.0) {
+					uncompressed_keys.push(compressed_key.serialize());
+				} else {
+					log::error!(target: "thea", "Unable to parse compressed key");
```
Comment (Gauthamastro) [inline pallets/thea/src/lib.rs:773]: Emit an event here.

**Review by Gauthamastro — DISMISSED** [review]

File: pallets/thea/src/tests.rs [hunk 2]
```diff
@@ -414,7 +414,7 @@ fn test_handle_misbehaviour_happy_path_invalid_proposal() {
 			fork_period: 0,
 			min_stake: 1_000_000,
 			fisherman_stake: 1_000_000,
-			key_type: KeyType::Compressed,
+			key_type: NetworkType::Parachain,
```
Comment (Gauthamastro) [inline pallets/thea/src/tests.rs]: Rename parameter

File: pallets/thea/src/tests.rs [hunk 3]
```diff
@@ -378,7 +378,7 @@ fn test_handle_misbehaviour_happy_path_valid_proposal() {
...
```
Comment (Gauthamastro) [inline pallets/thea/src/tests.rs]: Rename parameter

File: pallets/thea/src/tests.rs [hunk 4]
```diff
@@ -343,7 +343,7 @@ fn test_report_misbehaviour_not_enough_stake() {
 			fork_period: 0,
 			min_stake: 1_000_000_000_000_000_000_000_000_000,
 			fisherman_stake: 1_000_000_000_000_000_000_000_000,
-			key_type: KeyType::Compressed,
+			key_type: NetworkType::Parachain,
```
Comment (Gauthamastro) [inline pallets/thea/src/tests.rs]: same

**Review by Gauthamastro — CHANGES_REQUESTED** [review]

File: pallets/thea/src/tests.rs (hunk 4 above)
Comment (zktony) [reply pallets/thea/src/tests.rs]: done

**Review by Gauthamastro — APPROVED** [review]

**Review by serhii-temchenko — APPROVED** [review]

**Pull request merged**